### DISCUSSION TITLE
How to export HTML in nodejs while excluding huge files

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#6619] [Q&A] [answered]
I found when using nodejs wiki, exporting HTML using this button will download a 15M HTML, which is what I need

<img width="178" alt="截屏2022-04-11 15 34 03" src="https://user-images.githubusercontent.com/3746270/162687161-f79fae52-267f-4875-a787-db7448bb4d17.png">
 
But exporting HTML in the nodejs server-side will result in 250M+ HTML containing large PDF that I drag&drop to it.

https://github.com/tiddly-gittly/tw-mobile-sync/blob/33f8318f5e264c86c56d1a8186d104556fdb6a4e/src/server/server-get-html-endpoint.ts#L13-L20

How can I export HTML while excluding those large files? Only via modifying the export template?

### Additional Context

I want to export HTML and send it to the mobile client, because I need `document.write(fullHtml);` to load plugins from the server.

The filter I used for normal tiddler sync

```js
`[all[]] :filter[get[modified]compare:date:gt[${lastSync ?? ''}]]`
```

won't work for plugin that is older than the time `lastSync`.

---

I have a clue, which template does

> browsers now limit the size of data URIs which breaks large embedded PDFs
> https://github.com/Jermolene/TiddlyWiki5/discussions/6189#discussioncomment-1621727

use? I can use that template to save HTML, to get a HTML that is the same size as the one on the browser.

I'm using `root-tiddler=$:/core/save/lazy-images` to start the nodejs server, but I can see some image when open the downloaded snapshot, so this might not the template when serving. And I don't know how does it exclude the pdf...

**Comment by Jermolene** [comment]
Hi @linonetwo I think you'll have to define a custom export filter that excludes the PDF files, if that makes sense?

**Reply by linonetwo** [reply]
That is OK, but I just want to reuse some existing filters or templates if there are...

**Reply by linonetwo** [reply]
What does "browsers now limit the size of data URIs" use?

**Reply by linonetwo** [reply]
I see, its core/modules/server/routes/get-index.js

**Comment by linonetwo** [comment, marked as answer]
Just use lazy-image template, it will filter out PDFs.

The server side rendering code is very simple, just core/modules/server/routes/get-index.js

```
exports.handler = function(request,response,state) {
	var text = state.wiki.renderTiddler(state.server.get("root-render-type"),state.server.get("root-tiddler")),
		responseHeaders = {
		"Content-Type": state.server.get("root-serve-type")
	};
	state.sendResponse(200,responseHeaders,text);
};
```
